cloudformation.yaml

```
@@ -783,6 +783,89 @@ Resources:
       BatchSize: 1
 
 
+  AppleSubsStatus5xxErrors:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: !FindInMap [StageVariables, !Ref Stage, AlarmActionsEnabled]
+      AlarmActions:
+        - Ref: AlarmTopic
+      OKActions:
+        - Ref: AlarmTopic
+      AlarmName: !Sub mobile-purchases-${Stage}-apple-subscription-status-check-errors
+      AlarmDescription: |
+        More than 20% of attempts to check Apple subscription status resulted in a 5XX error.
+      Metrics:
+        - Id: e1
+          Label: Percentage of requests which result in a 5XX error
+          Expression: "100*(m1/m2)"
+        - Id: m1
+          Label: Number of 5XX responses
+          MetricStat:
+            Metric:
+              MetricName: 5XXError
+              Namespace: AWS/ApiGateway
+              Dimensions:
+                - Name: ApiName
+                  Value: !Sub ${App}-${Stage}
+                - Name: Method
+                  Value: POST
+                - Name: Resource
+                  Value: /apple/subscription/status
+                - Name: Stage
+                  Value: !Ref Stage
+            Period: 600
+            Stat: Sum
+          ReturnData: false
+        - Id: m2
+          Label: Total number of requests
+          MetricStat:
+            Metric:
+              MetricName: Count
+              Namespace: AWS/ApiGateway
+              Dimensions:
+                - Name: ApiName
+                  Value: !Sub ${App}-${Stage}
+                - Name: Method
+                  Value: POST
+                - Name: Resource
+                  Value: /apple/subscription/status
+                - Name: Stage
+                  Value: !Ref Stage
+            Period: 600
+            Stat: Sum
+          ReturnData: false
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      Threshold: 20
+
+  ApplePubsub5xxErrors:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: !FindInMap [StageVariables, !Ref Stage, AlarmActionsEnabled]
+      AlarmActions:
+        - Ref: AlarmTopic
+      OKActions:
+        - Ref: AlarmTopic
+      AlarmName: !Sub mobile-purchases-${Stage}-apple-pubsub-check-errors
+      AlarmDescription: A HTTP request to the pubsub endpoint resulted in a 5XX error.
+      Dimensions:
+        - Name: ApiName
+          Value: !Sub ${App}-${Stage}
+        - Name: Method
+          Value: POST
+        - Name: Resource
+          Value: /apple/pubsub
+        - Name: Stage
+          Value: !Ref Stage
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      MetricName: 5XXError
+      Namespace: AWS/ApiGateway
+      Period: 300
+      Statistic: Sum
+      Threshold: 1
```

An absence of errors is a good thing in this case so I think we should be [treating missing data](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html#cfn-cloudwatch-alarms-treatmissingdata) as `notBreaching`.

```suggestion
      Threshold: 1
      TreatMissingData: notBreaching
```